osu.Game/Overlays/ChatOverlayV2.cs

```diff
@@ -165,7 +165,12 @@ protected override void LoadComplete()
             };
             channelList.OnRequestLeave += channel => channelManager.LeaveChannel(channel);
 
-            channelListing.OnRequestJoin += channel => channelManager.JoinChannel(channel);
+            channelListing.OnRequestJoin += channel =>
+            {
+                channelManager.JoinChannel(channel);
+                // Manually joining a channel should keep the selector open
+                selectorActive.Value = true;
```

Comment:
This change is a bit awkward, but it follows that if the overlay should always show the current channel when the channel is set externally, manual input from user actions should have to preserve expected behaviour by setting the selector state like this.
It's not the most elegant way of doing it, but I'm not sure how else to go about it aside from manually setting the current channel to `null` anytime the selector is visible which I don't think is a good idea.

Reply:
Coming from a bit of a [discussion in discord](https://discord.com/channels/188630481301012481/188630652340404224/973345089764077578), here is what I think should happen to make this work without looking awkward like that:
 - `CurrentChannel.Value = null` indicates that no channel is currently selected.
 - Channel listing/selector visibility state should be bound to whether `CurrentChannel.Value == null`
 - Joining a channel should never select it for better UX, rather than what's shown in https://github.com/ppy/osu/pull/18181#discussion_r868353851.
   - All existing usages of `ChannelManager.JoinChannel` should be updated accordingly.

Would like more opinions from others, especially in whether `CurrentChannel.Value == null -> selector visible` makes sense.

Reply:
After rereading, I think using `CurrentChannel.Value = null` to indicate that the selector should be open is an OK approach.
However since it requires behaviour changes to the `ChannelManager` I'm a bit wary of implementing it as suggested so I will wait for more opinions.

Reply:
In the interest of simplicity, if `null` can mean that

- no channel is selected
OR
- the selector is visible

Then that is the best path forward.

Reply:
I've updated the overlay to show overlay when `CurrentChannel == null` which ends up simplifying it's logic overall.
To support this, I've added the optional parameter `setCurrent` to `ChannelManager.JoinChannel` to support this behaviour.
Additionally, the use of null makes the `SelectorActive Bindable<bool>` redundant in the `ChannelList` so I have removed its usage which also slightly simplifies it's logic.